tests/test_cli_client.cpp

```diff
@@ -1220,6 +1220,39 @@ TEST_F(Client, launch_cmd_mount_option)
     EXPECT_EQ(send_command({"launch", "--name", instance_name, "--mount", fake_source}), mp::ReturnCode::Ok);
 }
 
+TEST_F(Client, launchCmdMountOptionFailsOnInvalidDir)
+{
+    auto [mocked_file_ops, mocked_file_ops_guard] = mpt::MockFileOps::inject();
+    EXPECT_CALL(*mocked_file_ops, exists(A<const QFileInfo&>())).WillOnce(Return(false)).WillRepeatedly(Return(true));
+    EXPECT_CALL(*mocked_file_ops, isDir(A<const QFileInfo&>())).WillOnce(Return(false)).WillRepeatedly(Return(true));
+    EXPECT_CALL(*mocked_file_ops, isReadable(A<const QFileInfo&>()))
+        .WillOnce(Return(false))
+        .WillRepeatedly(Return(true));
+
+    const auto fake_source = "invalid/dir";
+    const auto fake_target = fake_source;
+    const auto instance_name = "some_instance";
+
+    std::stringstream err;
+    EXPECT_EQ(
+        send_command({"launch", "--name", instance_name, "--mount", fmt::format("{}:{}", fake_source, fake_target)},
+                     trash_stream, err),
+        mp::ReturnCode::CommandLineError);
+    EXPECT_THAT(err.str(), HasSubstr(fmt::format("Mount source path \"{}\" does not exist", fake_source)));
+
+    EXPECT_EQ(
+        send_command({"launch", "--name", instance_name, "--mount", fmt::format("{}:{}", fake_source, fake_target)},
+                     trash_stream, err),
+        mp::ReturnCode::CommandLineError);
+    EXPECT_THAT(err.str(), HasSubstr(fmt::format("Mount source path \"{}\" does not exist", fake_source)));
+
+    EXPECT_EQ(
+        send_command({"launch", "--name", instance_name, "--mount", fmt::format("{}:{}", fake_source, fake_target)},
+                     trash_stream, err),
+        mp::ReturnCode::CommandLineError);
+    EXPECT_THAT(err.str(), HasSubstr(fmt::format("Mount source path \"{}\" does not exist", fake_source)));
```

Comment:
hmm shouldn't these return different error messages? i think you should clear `err` before each `send_command` invocation, so you don't compare against errors from previous calls.

Reply:
Sorry, copy paste error! I also added a line to clear the stream to make sure that the previous output is gone.